src/main/java/com/iexec/worker/feign/CustomCoreFeignClient.java

```diff
@@ -129,21 +129,23 @@ public Optional<ReplicateTaskSummary> getAvailableReplicateTaskSummary(long last
     }
 
     public TaskNotificationType updateReplicateStatus(String chainTaskId, ReplicateStatusUpdate replicateStatusUpdate) {
-        Map<String, Object> arguments = new HashMap<>();
-        arguments.put(JWTOKEN, loginService.getToken());
-        arguments.put("chainTaskId", chainTaskId);
-        arguments.put("statusUpdate", replicateStatusUpdate);
-
-        HttpCall<TaskNotificationType> httpCall = args ->
-                coreClient.updateReplicateStatus((String) args.get(JWTOKEN), (String) args.get("chainTaskId"),
-                        (ReplicateStatusUpdate) args.get("statusUpdate"));
-
-        ResponseEntity<TaskNotificationType> response = makeHttpCall(httpCall, arguments, "updateReplicateStatus");
-        if (!is2xxSuccess(response)) {
+        try {
+            final ResponseEntity<TaskNotificationType> response = coreClient.updateReplicateStatus(
+                    loginService.getToken(),
+                    chainTaskId,
+                    replicateStatusUpdate
+            );
+            final TaskNotificationType notificationType = response.getBody();
+            log.info(replicateStatusUpdate.getStatus().toString() + " [chainTaskId:{}]", chainTaskId);
+            return notificationType;
+        } catch (FeignException.Unauthorized e) {
+            login();
```

Comment:
I think we should add a todo/note about non thread safety on top of `LoginService#login()`

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/499/commits/10ef597e8ec7e4dd2c4fbd4eef4dba353e90d5b2.